sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuInsertIntoHadoopFsRelationCommand.scala

```diff
@@ -64,6 +64,30 @@ case class GpuInsertIntoHadoopFsRelationCommand(
     val fs = outputPath.getFileSystem(hadoopConf)
     val qualifiedOutputPath = outputPath.makeQualified(fs.getUri, fs.getWorkingDirectory)
 
+    val parameters = CaseInsensitiveMap(options)
+
+    val partitionOverwriteMode = parameters.get("partitionOverwriteMode")
+      // scalastyle:off caselocale
+      .map(mode => PartitionOverwriteMode.withName(mode.toUpperCase))
+      // scalastyle:on caselocale
+      .getOrElse(sparkSession.sessionState.conf.partitionOverwriteMode)
+
+
+    val enableDynamicOverwrite = partitionOverwriteMode == PartitionOverwriteMode.DYNAMIC
+    // This config only makes sense when we are overwriting a partitioned dataset with dynamic
+    // partition columns.
+    val dynamicPartitionOverwrite = enableDynamicOverwrite && mode == SaveMode.Overwrite &&
+      staticPartitions.size < partitionColumns.length
+
+    val jobId = java.util.UUID.randomUUID().toString
+
+    val committerOutputPath = if (dynamicPartitionOverwrite) {
```

Comment:
nit, might be nice to pull the comment over:
 // For dynamic partition overwrite, FileOutputCommitter's output path is staging path, files
 // will be renamed from staging path to final output path during commit job

Reply:
nice, will do

Reply:
done.https://github.com/NVIDIA/spark-rapids/pull/7379/commits/d8f2a1ef0b0aa0076f37c8ef4dd840045e52ae8c